src/decisionengine/framework/engine/Workers.py

```diff
@@ -51,60 +50,42 @@ def get_consumes(self):
+        unit_testing = self.logger_config.get("unit_testing", "False")
 
-        self.logger = structlog.getLogger(CHANNELLOGGERNAME)
+        self.logger = structlog.getLogger(logconf.CHANNELLOGGERNAME)
 
         # setting a default value here. value from config file is set in call
         # self.task_manager.set_loglevel_value after logger configuration is completed
-        logging.getLogger(CHANNELLOGGERNAME).setLevel(logging.DEBUG)
+        logging.getLogger(logconf.CHANNELLOGGERNAME).setLevel(logging.DEBUG)
 
-        # TODO:
-        # alter decisionengine.framework.modules.de_logger set_logging
-        # so we can reuse it here and reduce duplication
         logger_rotate_by = self.logger_config.get("file_rotate_by", "size")
-
-        # logconf is our global object edited in global space on purpose
-
         if logger_rotate_by == "size":
-            logconf["handlers"].update(
-                {
-                    myname: {
-                        "level": "DEBUG",
-                        "filename": myfilename,
-                        "formatter": "plain",
-                        "class": "logging.handlers.RotatingFileHandler",
-                        "maxBytes": MAX_CHANNEL_FILE_SIZE,
-                        "backupCount": self.logger_config.get("max_backup_count", 6),
-                    }
-                }
+            handler = logging.handlers.RotatingFileHandler(
+                filename=myfilename,
+                maxBytes=self.logger_config.get("max_file_size", 200 * MB),
+                backupCount=self.logger_config.get("max_backup_count", 6),
             )
+
         elif logger_rotate_by == "time":
-            logconf["handlers"].update(
-                {
-                    myname: {
-                        "level": "DEBUG",
-                        "filename": myfilename,
-                        "formatter": "plain",
-                        "class": "logging.handlers.TimedRotatingFileHandler",
-                        "when": "D",
-                        "interval": 1,
-                    }
-                }
+            handler = logging.handlers.TimedRotatingFileHandler(
+                filename=myfilename,
+                when=self.logger_config.get("rotation_time_unit", "D"),
+                interval=self.logger_config.get("rotation_time_interval", 1),
+                backupCount=self.logger_config.get("max_backup_count", 6),
             )
         else:
             raise ValueError(f"Incorrect 'logger_rotate_by':'{logger_rotate_by}:'")
 
-        logconf["loggers"].update(
-            {
-                CHANNELLOGGERNAME: {
-                    "handlers": [myname, "file_structlog_debug"],
-                }
-            }
-        )
-        logging.config.dictConfig(logconf)
+        handler.setLevel(logging.DEBUG)
+        handler.setFormatter(logging.Formatter(logconf.userformat))
+
+        self.logger.addHandler(handler)
+
+        if unit_testing == "False":
```

Comment:
When set to `False` my local instance locks up on the second message sent to the log.  May be a local bug in my setup, but figured you'd want to know.

Reply:
I don't have this problem. I have tested job_classification channel and resource_request channel and get full logs (until an error from the latter channel).